packages/calcite-components/src/components/stepper/stepper.tsx

```diff
@@ -310,25 +330,50 @@ export class Stepper implements LocalizedComponent, T9nComponent {
     updateMessages(this, this.effectiveLocale);
   }
 
-  private itemMap = new Map<HTMLCalciteStepperItemElement, { position: number; content: Node[] }>();
+  /** keep track of the currently active item position */
+  @State() currentPosition: number;
 
-  /** list of sorted Stepper items */
-  private items: HTMLCalciteStepperItemElement[] = [];
+  @Watch("currentPosition")
+  handlePositionChange(): void {
+    requestAnimationFrame(() => {
```

Comment:
Since `determineActiveStepper` is calling `getComputedStyle`, could you try using [`readTask`](https://stenciljs.com/docs/api#other) vs `requestAnimationFrame` here?